engine/schema/src/com/cloud/storage/dao/StoragePoolTagsDaoImpl.java

```diff
@@ -77,4 +92,71 @@ public void deleteTags(long poolId) {
         txn.commit();
     }
 
+    @Override
+    public List<StoragePoolTagVO> searchByIds(Long... stIds) {
+        final int detailsBatchSize = getDetailsBatchSize();
+
+        // query details by batches
+        List<StoragePoolTagVO> uvList = new ArrayList<StoragePoolTagVO>();
+        int curr_index = 0;
+
+        if (stIds.length > detailsBatchSize) {
```

Comment:
Correct me if I am wrong, but this `if` is not needed.
Let's assume the configuration:
```
batchSize=2000
current_index=0
lengthOfStIds=100
```

`curr_index + detailsBatchSize = 0 + 2000`, which is not less than the size of the array `(100)`. Therefore, the while is not executed. Then, the pools will be loaded at lines 100-112